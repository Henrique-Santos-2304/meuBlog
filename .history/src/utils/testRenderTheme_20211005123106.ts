import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(<ThemeProvider>{children}</ThemeProvider> );
};
