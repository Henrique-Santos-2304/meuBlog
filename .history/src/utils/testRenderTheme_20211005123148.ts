import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactElement): RenderResult => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider> );
};
