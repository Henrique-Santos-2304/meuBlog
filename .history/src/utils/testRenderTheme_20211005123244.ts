import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider> );
};
