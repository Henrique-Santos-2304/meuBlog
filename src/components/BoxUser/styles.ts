import styled, { css } from "styled-components";

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin-right: 8rem;

  @media (max-width: 1020px) {
    margin: -1rem -0.5rem 0 0;
  }
  @media (max-width: 480px) {
    margin: -2rem -2rem 0 0;
  }
`;

export const ImageLogo = styled.img`
  width: 20rem;
  height: 20rem;
  border-radius: 50%;
  box-shadow: 0 0 10px ${({ theme }) => theme.colors.red800},
    0 0 10px ${({ theme }) => theme.colors.white};
  @media (max-width: 780px) {
    width: 11rem;
    height: 11rem;
  }
  @media (max-width: 480px) {
    width: 8rem;
    height: 8rem;
  }
`;

export const Heading = styled.h1`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    font-family: ${theme.font.family.Tangerine};
    font-size: 5rem;
    margin-top: 2rem;
    @media (max-width: 1020px) {
      font-size: 3rem;
    }
    @media (max-width: 480px) {
      font-size: 4rem;
    }
  `}
`;
