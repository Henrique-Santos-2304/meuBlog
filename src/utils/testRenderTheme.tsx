import { ThemeProvider } from "styled-components";
import { ChakraProvider } from "@chakra-ui/react";

import { render, RenderResult } from "@testing-library/react";
import theme from "styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeProvider theme={theme}>
      <ChakraProvider> {children}</ChakraProvider>
    </ThemeProvider>
  );
};
