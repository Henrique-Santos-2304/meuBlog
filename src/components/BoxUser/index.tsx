import * as S from "./styles";

const BoxUser = () => (
  <S.Wrapper>
    <S.ImageLogo
      src="https://res.cloudinary.com/defnibbpl/image/upload/v1633360611/medium_henrique_techs_89a8aae1bc.jpg"
      alt="Foto Programador"
    ></S.ImageLogo>
    <S.Heading>Henrique Dos Santos</S.Heading>
  </S.Wrapper>
);

export default BoxUser;
