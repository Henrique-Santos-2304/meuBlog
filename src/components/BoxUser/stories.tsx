import { Story, Meta } from "@storybook/react";
import BoxUser from ".";

export default {
  title: "BoxUser",
  component: BoxUser,
} as Meta;

export const Basic: Story = (args) => <BoxUser {...args} />;

Basic.args = {
  title: "title basic",
  description: "description basic",
};

export const Default: Story = (args) => <BoxUser {...args} />;
