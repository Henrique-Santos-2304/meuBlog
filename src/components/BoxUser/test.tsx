import { renderTheme } from "utils/testRenderTheme";
import BoxUser from ".";
describe("BoxUser", () => {
  it("should render ", () => {
    const { container } = renderTheme(<BoxUser />);
    expect(container).toMatchSnapshot();
  });
});
