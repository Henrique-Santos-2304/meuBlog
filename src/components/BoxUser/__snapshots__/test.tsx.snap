// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxUser should render  1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 8rem;
}

.c1 {
  width: 20rem;
  height: 20rem;
  border-radius: 50%;
  box-shadow: 0 0 10px #C11010,0 0 10px #E5E7EB;
}

.c2 {
  color: #E5E7EB;
  font-family: 'Tangerine',cursive;
  font-size: 5rem;
  margin-top: 2rem;
}

@media (max-width:1020px) {
  .c0 {
    margin: -1rem -0.5rem 0 0;
  }
}

@media (max-width:480px) {
  .c0 {
    margin: -2rem -2rem 0 0;
  }
}

@media (max-width:780px) {
  .c1 {
    width: 11rem;
    height: 11rem;
  }
}

@media (max-width:480px) {
  .c1 {
    width: 8rem;
    height: 8rem;
  }
}

@media (max-width:1020px) {
  .c2 {
    font-size: 3rem;
  }
}

@media (max-width:480px) {
  .c2 {
    font-size: 4rem;
  }
}

<div>
  <div
    class="c0"
  >
    <img
      alt="Foto Programador"
      class="c1"
      src="https://res.cloudinary.com/defnibbpl/image/upload/v1633360611/medium_henrique_techs_89a8aae1bc.jpg"
    />
    <h1
      class="c2"
    >
      Henrique Dos Santos
    </h1>
  </div>
</div>
`;
